galaxy_ng/app/api/ui/serializers/collection.py

```diff
@@ -112,40 +112,36 @@ class _CollectionSerializer(Serializer):
     name = serializers.CharField()
     download_count = serializers.IntegerField(default=0)
     latest_version = serializers.SerializerMethodField()
-    deprecated = serializers.SerializerMethodField()
 
     def get_namespace(self, obj):
         namespace = Namespace.objects.get(name=obj.namespace)
         return NamespaceSummarySerializer(namespace).data
 
-    def get_deprecated(self, obj):
-        return obj.collection.deprecated
-
-    def _get_versions_in_repo(self, obj):
-        distro = AnsibleDistribution.objects.get(base_path=self.context['path'])
-        repository_version = distro.repository.latest_version()
-        versions_in_repo = CollectionVersion.objects.filter(
-            pk__in=repository_version.content,
-            collection=obj.collection,
-        )
-        return sorted(
-            versions_in_repo, key=lambda obj: semantic_version.Version(obj.version), reverse=True
-        )
-
 
 class CollectionListSerializer(_CollectionSerializer):
+    deprecated = serializers.BooleanField()
+
     def get_latest_version(self, obj):
         return CollectionVersionBaseSerializer(obj).data
 
 
 class CollectionDetailSerializer(_CollectionSerializer):
     all_versions = serializers.SerializerMethodField()
 
-    def get_all_versions(self, obj):
-        versions_in_repo = self._get_versions_in_repo(obj)
-        return CollectionVersionSummarySerializer(versions_in_repo, many=True).data
-
-    # TODO(awcrosby): rename field to "version_details" since with
+    # TODO: rename field to "version_details" since with
     # "version" query param this won't always be the latest version
     def get_latest_version(self, obj):
         return CollectionVersionDetailSerializer(obj).data
+
+    def get_all_versions(self, obj):
+        path = self.context['request'].parser_context['kwargs']['path']
+        distro = AnsibleDistribution.objects.get(base_path=path)
+        repository_version = distro.repository.latest_version()
+        versions_in_repo = CollectionVersion.objects.filter(
+            pk__in=repository_version.content,
+            collection=obj.collection,
+        )
+        versions_in_repo = sorted(
+            versions_in_repo, key=lambda obj: semantic_version.Version(obj.version), reverse=True
```

Comment:
I've mentioned it in discussions before, but I'm a little worried that galaxy and ansible-galaxy may not be using the same version compare. Hopefully we can get away with pure semver, and ansible-galaxy's vercmp will handle it correctly as a subset.

Not much to do about that, and I'm assuming this is more or less based on the pulp_ansible equilivent?

Reply:
I think this part is not based on pulp_ansible, but used to list all versions to populate our GUI dropdown. For this particular part it looks like it's using semver to sort rather than compare.